File: src/AnalistaFinanziarioIA.Core/Models/Titolo.cs
namespace AnalistaFinanziarioIA.Core.Models;

public enum TipoTitolo { Azione, ETF, BTP, Crypto, ObbligazioneCorp }

public class Titolo
{
    public int Id { get; set; }
    public string Simbolo { get; set; } = string.Empty; // Es: VWCE
    public string? Isin { get; set; }                   // Es: IE00BK5BQT80
    public string Nome { get; set; } = string.Empty;
    public string? Settore { get; set; }
    public string? Mercato { get; set; }                // Es: Borsa Italiana, XETRA
    public string Valuta { get; set; } = "EUR";         // Fondamentale per i calcoli!
    public DateTime DataCreazione { get; set; } = DateTime.Now;
    public TipoTitolo Categoria { get; set; }

    public decimal UltimoPrezzo { get; set; }

    public ICollection<QuotazioneStorica> QuotazioniStoriche { get; set; } = new List<QuotazioneStorica>();
    public ICollection<AnalisiFinanziaria> Analisi { get; set; } = new List<AnalisiFinanziaria>();
}

File: src/AnalistaFinanziarioIA.API/Scripts/Models/Titolo.ts
/**
 * Rispecchia l'Enum C# AnalistaFinanziarioIA.Core.Models.TipoTitolo
 */
export enum TipoTitolo {
    Azione = 0,
    ETF = 1,
    BTP = 2,
    Crypto = 3,
    ObbligazioneCorp = 4
}

// Corrisponde a: public enum TipoOperazione (assunto dai DTO)
export enum TipoOperazione {
    Acquisto = 0,
    Vendita = 1
}

/**
 * Rispecchia la classe C# AnalistaFinanziarioIA.Core.Models.Titolo
 */
export interface ITitolo {
    id: number;
    simbolo: string;
    isin?: string;
    nome: string;
    settore?: string;
    mercato?: string;
    valuta: string;
    dataCreazione: string; // DateTime -> string (ISO)
    tipo: TipoTitolo;
    ultimoPrezzo: number;
    dataUltimoPrezzo: string;
}